<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="GazeboSystem" type="system">
        <hardware>
            <plugin>gazebo_ros2_control/GazeboSystem</plugin>
        </hardware>
        <joint name="base_link1_joint">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>
        <joint name="link1_link2_joint">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>
        <joint name="link2_link3_joint">
            <command_interface name="position">
                <param name="min">-0.523</param>
                <param name="max">2.61</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>
    </ros2_control>

    <gazebo>
        <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
            <parameters>$(find quadruped_description)/config/my_controllers.yaml</parameters>
        </plugin>
    </gazebo>

</robot>